### PR TITLE
Simplicial set group

### LINKED ISSUE #34886: Implement finite covers of simplicial sets
<div id="comment:0"></div>

This branch implements the computation of finite covers of simplicial sets, and also the simplicial set associated to a grpup presentation.

CC:  @jhpalmieri

Component: **algebraic topology**

Author: **Miguel Marco**

_Issue created by migration from https://trac.sagemath.org/ticket/34886_

### PR DESCRIPTION
**Pull request by miguelmarco**
### 📚 Implementation of finite covers of simplicial sets

Compute the cover of a (based) simplicial set associated to a representation of its fundamental group to a finite group.

Fixes #34886.

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ ] I have linked an issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/topology/simplicial_set_examples.py [hunk 1]
```diff
@@ -790,3 +794,65 @@ def HopfMap():
...
                        alpha_3:s2_sigma, alpha_4:s0_sigma,
                        alpha_5:s2_sigma, alpha_6:s1_sigma})
+
+
+def PresentationComplex(G):
+    r"""
+    Return a simplicial set constructed from a group presentation.
+    The result is a subdivision of the presentation complex.
+
+    The presentation complex has one vertex and one edge for
```
Comment (jhpalmieri) [inline src/sage/topology/simplicial_set_examples.py]:
```suggestion
    The presentation complex has a single vertex and it has one edge for
```
(It is possible to read the current version as meaning one vertex for each generator and one edge for each generator.)

File: src/sage/categories/simplicial_sets.py [hunk 2]
```diff
@@ -361,10 +369,197 @@ def fundamental_group(self, simplify=True):
...
+                if not edges:
+                    return self.identity()
+                G, char = self._universal_cover_dict()
+                return self.covering_map(char)
+
+            def covering_map(self, character):
+                r"""
+                Return the covering map associated to a character.
+
+                The character is represented by a dictionary, that assigns an
```
Comment (jhpalmieri) [inline src/sage/categories/simplicial_sets.py]:
```suggestion
                The character is represented by a dictionary that assigns an
```

File: src/sage/categories/simplicial_sets.py [hunk 3]
```diff
@@ -331,19 +331,27 @@ def fundamental_group(self, simplify=True):
...
+                FG = self._universal_cover_dict()[0]
+                if simplify:
+                    return FG.simplified()
+                else:
+                    return FG
 
+            def _universal_cover_dict(self):
+                r"""
+                Return the fundamental group and dictionary sending each edge to
+                the corresponding group element
```
Comment (jhpalmieri) [inline src/sage/categories/simplicial_sets.py:345]: This method needs doctests.

File: src/sage/categories/simplicial_sets.py [hunk 4]
```diff
@@ -361,10 +369,197 @@ def fundamental_group(self, simplify=True):
...
+                     (1, e): None,
+                     (f, 1): ((1, e), (1, 1)),
+                     (f, e): ((1, 1), (1, e)),
+                     (f * f, 1): ((f, e), s_0 (1, 1), (f, 1)),
+                     (f * f, e): ((f, 1), s_0 (1, e), (f, e)),
+                     (f * f * f, 1): ((f * f, e), s_0 (f, 1), s_1 (f, 1), (f * f, 1)),
+                     (f * f * f, e): ((f * f, 1), s_0 (f, e), s_1 (f, e), (f * f, e))}
+                    sage: C.fundamental_group()
+                    Finitely presented group <  |  >
+
```
Comment (jhpalmieri) [inline src/sage/categories/simplicial_sets.py]: I suggest deleting these two blank lines.

**Review by jhpalmieri — CHANGES_REQUESTED** [review]
Overall it looks very good to me. I've suggested a few very minor changes, and one method needs doctests. Then it will be ready to go.

**Comment by miguelmarco** [comment]
Thanks for the review. I did the suggested changes.

**Comment by jhpalmieri** [comment]
Looks great, thank you!

**Review by jhpalmieri — APPROVED** [review]

**Pull request merged**